wicker/core/storage.py

```diff
@@ -215,64 +215,115 @@ class S3PathFactory:
+        """Init the path factory.
+
+        Object to form the expected paths and return them to the user based of root path and storage bool.
+
+        Args:.
+            root_path (str): File system loc of the root of the wicker file structure.
+            store_concatenated_bytes_files_in_dataset (bool, optional): Whether to assume concat bytes files are stored.
+                Defaults to False
+        """
+        self.root_path: str = root_path
+        self.store_concatenated_bytes_files_in_dataset = store_concatenated_bytes_files_in_dataset
 
     def __eq__(self, other: Any) -> bool:
         return super().__eq__(other) and type(self) == type(other) and self.root_path == other.root_path
 
-    def get_dataset_assets_path(self, dataset_id: DatasetID, s3_prefix: bool = True) -> str:
+    def _get_dataset_assets_path(self, dataset_id: DatasetID, prefix: Optional[str] = None) -> str:
+        """Get the asset path in known file structure.
+
+        Args:
+            dataset_id (DatasetID): Id of the dataset
+            prefix (Optional[str], optional): Optional prefix to remove from file paths. Defaults to None.
+
+        Returns:
+            str: path to assets folder
+        """
         full_path = os.path.join(
             self.root_path,
             dataset_id.name,
             dataset_id.version,
             "assets",
         )
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_dataset_partition_metadata_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+    def _get_dataset_partition_metadata_path(
+        self, data_partition: DatasetPartition, prefix: Optional[str] = None
+    ) -> str:
+        """Get partition metadata path in wicker known file structure.
+
+        Args:
+            data_partition (DatasetPartition): data partition to use for path
+            prefix (Optional[str], optional): Optional prefix to remove from file paths. Defaults to None.
+
+        Returns:
+            str: Path to partition metadata json file.
+        """
         full_path = os.path.join(
-            self.get_dataset_partition_path(data_partition),
+            self._get_dataset_partition_path(data_partition),
             "_l5ml_dataset_partition_metadata.json",
         )
-        if not s3_prefix:
-            return full_path.replace("s3://", "")
+        if prefix:
+            return full_path.replace(prefix, "")
         return full_path
 
-    def get_dataset_partition_path(self, data_partition: DatasetPartition, s3_prefix: bool = True) -> str:
+    def _get_dataset_partition_path(self, data_partition: DatasetPartition, prefix: Optional[str] = None) -> str:
+        """Get the dataset partition parquet path.
+
+        Private gettr handling logic of pathing centrally.
```

Comment:
Nit: typo/spelling: `gettr` -> `getter`.
```suggestion
        Private getter handling logic of pathing centrally.
```